rust/rope/src/engine.rs

```diff
@@ -632,16 +633,38 @@ fn compute_deltas(revs: &[Revision], text: &Rope, tombstones: &Rope, deletes_fro
     out
 }
 
+/// Computes a series of priorities and transforms for the deltas on the right
+/// from the new revisions on the left.
+///
+/// Applies an optimization where it combines sequential revisions with the
+/// same priority into one transform to decrease the number of transforms that
+/// have to be considered in `rebase` substantially for normal editing
+/// patterns. Any large runs of typing in the same place by the same user (e.g
+/// typing a paragraph) will be combined into a single segment in a transform
+/// as opposed to thousands of revisions.
+fn compute_transforms(revs: Vec<Revision>) -> Vec<(usize, Subset)> {
+    let mut out = Vec::new();
+    let mut last_priority: Option<usize> = None;
+    for r in revs.into_iter() {
+        if let Contents::Edit {priority, inserts, .. } = r.edit {
```

Comment:
A question: since this seems to only be aggregating the inserts, would it be possible to skip over revs with empty insert sets? That would reduce the number of `transform_union` calls, and also (potentially) aggregate more if you had deletes with one priority interleaved with inserts of another.